### PR TITLE
feat: add Redoc to Redocly CDN

### PR DESCRIPTION
**Pull request by AlexVarchuk**
## What/Why/How?
 - add Redoc to Redocly CDN
 - create CI/CD for automation deployment
 - update docs
 
## Check yourself

- [x] Code is linted
- [x] Tested
- [x] All new/updated code is covered with tests

### CODE REVIEW

File: scripts/publish-cdn.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+else
+  echo Releasing $VERSION
+
+  echo Uploading to S3 $VERSION
+  aws s3 cp bundles "s3://redocly-cdn/redoc/v$VERSION/bundles" --recursive "$@"
+  aws s3 cp typings "s3://redocly-cdn/redoc/v$VERSION/typings" --recursive "$@"
+  aws s3 cp CHANGELOG.md "s3://redocly-cdn/redoc/v$VERSION/CHANGELOG.md" "$@"
+  aws s3 cp LICENSE "s3://redocly-cdn/redoc/v$VERSION/LICENSE" "$@"
+  aws s3 cp package.json "s3://redocly-cdn/redoc/v$VERSION/package.json" "$@"
+  aws s3 cp README.md "s3://redocly-cdn/redoc/v$VERSION/README.md" "$@"
```
Comment (BohdanPetryshyn) [inline scripts/publish-cdn.sh:22]: We might want to extract these similar pieces of script into [bash functions](https://ryanstutorials.net/bash-scripting-tutorial/bash-functions.php).

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -42,7 +44,7 @@ jobs:
       - name: Download bundled artifact
         uses: actions/download-artifact@v3
         with:
-          name: bundles
+          name: publish
```
Comment (BohdanPetryshyn) [inline .github/workflows/publish.yml:47]: In my opinion, the name of the artifact should be a noun and represent the artifact's content. I think the name `bundle` makes sense for this artifact.

File: scripts/publish-cdn.sh (hunk 1 above)
Comment (AlexVarchuk) [reply scripts/publish-cdn.sh]: Good idea. done

**Review by BohdanPetryshyn — APPROVED** [review]

**Review by RomanHotsiy — APPROVED** [review]
LGTM, updated secrets too.

**Review by Oprysk — APPROVED** [review]

**Pull request merged**